x/concentrated-liquidity/position_test.go

```diff
@@ -1113,82 +1113,175 @@ func (s *KeeperTestSuite) TestMintSharesLockAndUpdate() {
 }
 
 func (s *KeeperTestSuite) TestPositionToLockCRUD() {
-	// Init suite for each test.
-	s.SetupTest()
-	s.Ctx = s.Ctx.WithBlockTime(DefaultJoinTime)
+	defaultLockDuration := 24 * time.Hour
+	clPool := s.PrepareConcentratedPool()
 	owner := s.TestAccs[0]
-	remainingLockDuration := 24 * time.Hour
 	defaultPositionCoins := sdk.NewCoins(DefaultCoin0, DefaultCoin1)
 
-	// Create a default CL pools
-	clPool := s.PrepareConcentratedPool()
-
-	// Fund the owner account
-	s.FundAcc(owner, defaultPositionCoins)
+	type positionToLockTest struct {
+		name                               string
+		createPosition                     func(s *KeeperTestSuite) (uint64, uint64)
+		expectedHasActiveLock              bool
+		expectedLockError                  bool
+		expectedPositionLockID             uint64
+		expectedPositionLockIDErr          bool
+		expectedGetPositionLockIdErr       bool
+		expectedUpdateHasActiveLock        bool
+		expectedUpdateLockID               uint64
+		expectedUpdateLockIDErr            bool
+		expectedUpdateGetPositionLockIdErr bool
```

Comment:
I'm really unsure how to reason about all the edge cases when there are so many conditions and so many things being tested.

I gave this my best shot but there are too many things going on to be able to follow the logic IMO. I would like to request separating these as much as possible, please. Ideally, each method should have an individual unit test testing all of its edge cases.

Reply:
Split up tests! https://github.com/osmosis-labs/osmosis/pull/5042/commits/6483d45cf7fab6975693c851ebe2af84f59b1a37